src/context/index.ts

```diff
@@ -1,27 +1,22 @@
 import { Request } from 'express';
-import { Session, SessionData } from 'express-session';
 import { request } from '../algorithm';
 
 export interface Context {
-  // TODO: keep here until we have login implemented. a partial given it could be undefined
-  session: Session & Partial<SessionData>;
-  // TODO: add context (aka. prisma, auth, etc.)
-  logout: () => Promise<void>;
-  // dependencies
+  req: any;
+  res: any;
   algorithm: typeof request;
 }
 
 export async function createContext({
   req,
+  res,
 }: {
   req: Request;
+  res: any;
 }): Promise<Context> {
```

Comment:
Under here I think it makes sense to check for the request headers and verify the token and fetch user details etc.

Reply:
You could even nest it into a `session` object so you can keep all existing code

Reply:
I still have to pass req